contents/performance/WebPerformance.md

```diff
@@ -0,0 +1,79 @@
+- Conversely, reducing page load times have been shown to have significant impact on [improving user engagement](https://medium.com/carousell-insider/how-we-made-carousells-mobile-web-experience-3x-faster-bbb3be93e006).
+- In addition, search engines have begun to use page load times as a [factor in determining search rankings](https://webmasters.googleblog.com/2018/01/using-page-speed-in-mobile-search.html).
+- Improving web performance is also essential in making your web application [accessible to users from emerging markets](https://building.calibreapp.com/beyond-the-bubble-real-world-performance-9c991dcd5342), where low-end devices and limited bandwidth are the norm.
+
+In order to deliver a positive user experience, web developers must ensure that their applications meet acceptable performance standards.
+
+## Key Ideas in Improving Web Performance
+There are many factors which affect web performance. Here, we give an overview of some of the key ideas that can be used to improve the performance of your web application.
+
+**Reduce Javascript Payloads**
+
+Loading and executing Javascript is often [the slowest part](https://medium.com/@addyosmani/the-cost-of-javascript-in-2018-7d8950fbb5d4) of the page load process. Reducing the amount of Javascript that needs to be loaded thus significantly reduces the time taken for your site to load.
+
+Here are some steps that web developers can take to reduce the amount of Javascript that clients need to load:
+- Use [code splitting](https://developers.google.com/web/fundamentals/performance/optimizing-javascript/code-splitting/) to load only the Javascript required for the page being accessed.
+- Remove unused code (often parts of libraries that are not used in the application) with [tree shaking](https://developers.google.com/web/fundamentals/performance/optimizing-javascript/tree-shaking/).
+- Ensure that the delivered Javascript is [minified](https://developers.google.com/web/fundamentals/performance/optimizing-content-efficiency/javascript-startup-optimization/).
+- Create [performance budgets](https://infrequently.org/2017/10/can-you-afford-it-real-world-web-performance-budgets/) that specify the maximum size of Javascript payloads, and [enforce them](https://web.dev/fast/incorporate-performance-budgets-into-your-build-tools) to ensure that the size of your Javascript payloads remain under control.
+
+**Optimize Images**
+
+Images form a [significant portion](https://httparchive.org/reports/page-weight) of the resources loaded on web applications. To improve performance and speed up the site for users, web developers should thus try to reduce the amount of bandwidth used to load images.
+
+Images can be optimized for the web in the following ways:
+- Use the [correct file format](https://medium.com/beginners-guide-to-mobile-web-development/web-image-formats-googles-webp-17e2fe5fc53e) for your images.
```

Comment:
Major issue: Reduce linking to other resources from main text as it transfers the burden to the reader to go and find the relevant paragraph from a long article. The user should not have to read external articles to understand this article. Instead, you can provide the relevant point as a popover/modal/sidebar and cite the source. In this case, explain succinctly why the correct file type needs to be used.
That is, one more layer of information is needed to make this article self-explanatory. Links to external sources should be for _truly_ optional info.